README.md

```diff
@@ -2,27 +2,204 @@
+
+4. If you choose `json` for `output_format` skip to step 5 otherwise see 4a
+
+    Example CEF template
+
+        [alerts_template]
+        template = "{{datetime_utc}} localhost CEF:1|{{vendor}}|{{product}}|{{product_version}}|{{reason_code}}|{{reason}}|{{severity}}|{{extension}}"
+        type_field = "type"
+        time_format = "%b %d %Y %H:%m:%S"
+        time_fields = ["backend_timestamp"]
+
+        [alerts_template.extension]
+        default = "cat={{type}}\tact={{sensor_action}}\toutcome={{run_state}}"
+        CB_ANALYTICS = "cat={{type}}\tact={{sensor_action}}\toutcome={{run_state}}\tframeworkName=MITRE_ATT&CK\tthreatAttackID={{attack_tactic}}:{{attack_technique}}"
+
+        [audit_logs_template]
+        template = "{{datetime_utc}} localhost CEF:1|{{vendor}}|{{product}}|{{product_version}}|Audit Logs|{{description}}|1|{{extension}}"
+        time_format = "%b %d %Y %H:%m:%S"
+        time_fields = ["eventTime"]
+
+        [audit_logs_template.extension]
+        default = "rt={{eventTime}}\tdvchost={{orgName}}\tduser={{loginName}}\tdvc={{clientIp}}\tcs4Label=Event_ID\tcs4={{eventId}}"
+
+    a. You'll need to create a template for each data type you plan to enable
+
+    b. Each data template supports a base `template` along with the option to specify an `extension` which can be used customize each message based on the values of the specified `type_field`
+
+    In the example above the `type_field` for alerts is set to `type` which enables a different extension to be selected based on the alert field `type`
+
+    **Note:** _If a value is not specified in the extension then the default option will be used. The values are CASE_SENSITIVE_
+
+    c. If you need to modify the format of a timestamp then you can specify a python strftime format in `time_format` as well as the `time_fields` that need to be modified
+
+    For more information on strftime formats see https://strftime.org/
 
-### Using Docker
+    d. See [Search Fields - Alert](https://developer.carbonblack.com/reference/carbon-black-cloud/platform/latest/alert-search-fields/) for the full list of Alert fields
+
+5. Based on the `output_type` you have choosen you'll need to configure one of the following output destinations
+
+    Examples outputs
+
+        file_path = "/some/dir"
+
+        http_out = https://example.com
+        http_headers =  "{ \"content-type\": \"application/json\" }"
+        https_ssl_verify = true
+
+        tcp_out = 1.2.3.5:514
+
+        udp_out = 1.2.3.5:514
+
+
+    a.  If you selected `tcp+tls` you'll need to configure the `tls` section based on your destination's expected certs
+
+        [tls]
+        ca_cert =
+        cert =
+        key =
+        key_password =
+        tls_verify =
+
+6. Configure one or more Carbon Black Cloud Organizations
+
+    Example Organization
+
+        [OrgName1]
+        server_url = defense.conferdeploy.net
+        org_key = ABCD1234
```

Comment:
How hard would it be to read from the credentials file for server, org key, api id & key?

Reply:
Not something I would like to approach as it causes too much confusion across multiple configuration locations and requires the profile name to be provided. I see an advanced user enjoying the benefit but this tool is targeted towards beginners